portfolioLogic.php

```diff
@@ -0,0 +1,120 @@
+<?php
+$db = new PDO('mysql:host=127.0.0.1; dbname=portfolioKyam', 'root');
+$db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+
+/**
+ * Gets selected fields from database of first portfolio item
+ *
+ * @param $db PDO to select from
+ * @return array of fields for item
+ */
+function getFirstPfItem(PDO $db) {
+    $query=$db->prepare("SELECT `title`,`description`,`imgRef`,`github`,`projURL`,`images`.`url`,`images`.`altText` 
+                         FROM `portfolioItems` 
+                         LEFT JOIN `images`
+                         ON `portfolioItems`.`imgRef`
+                         =`images`.`id`
+                         WHERE `portfolioItems`.`deleted` !=1 LIMIT 1;");
+    $query->execute();
+    $result= $query->fetch();
+    return $result;
+}
+
+/**
+ * Takes first portofolio item fields and creates an html section
+ *
+ * @param $result Array of getFirstPfItem function
+ *
+ * @return string of hmtl including item fields
+ */
+function createFirstPfItem(array $result):string {
+    return "<article class='primaryPfItem'>
+				<section class='itemPic'>
+					<img src=" . $result['url'] . " alt=" . $result['altText'] . "/>
+				</section>
+				<section class='itemText'>
+					<h3>
+						<a href='" . $result['projURL'] . "'>" . $result['title'] . "</a>
+					</h3>
+					<p>" . $result['description'] . "
+					</p>
+				</section>
+			</article>";
+}
+
+/**
+ * Gets selected fields from database of non-first portfolio items
+ *
+ * @param $db PDO to select from
+ * @return array of fields for items
+ */
+function getNonFirstPfItem(PDO $db) {
+    $query=$db->prepare("SELECT `title`,`description`,`imgRef`,`github`,`projURL`,`images`.`url`,`images`.`altText`
+                         FROM `portfolioItems`
+                         LEFT JOIN `images`
+                         ON `portfolioItems`.`imgRef`
+                         =`images`.`id`
+                         WHERE `portfolioItems`.`deleted` !=1  LIMIT 100 offset 1;"); //limit set as needed offset
+    $query->execute();
+    $result= $query->fetchAll();
+    return $result;
+}
+
+
+/**
+ * Takes non-first portofolio items fields and creates an html section
+ *
+ * @param $result Array result of getNonFirstPfItem function
+ *
+ * @return string of hmtl including items fields
```

Comment:
is hmtl a new language u created? i think thats a bit beyond the scope of this story

Reply:
resolved